packages/aws-rfdk/lib/deadline/lib/configure-spot-event-plugin.ts

```diff
@@ -0,0 +1,652 @@
+ *   - AWSThinkboxDeadlineSpotEventPluginAdminPolicy
+ *   - AWSThinkboxDeadlineResourceTrackerAdminPolicy
+ *   - A policy to pass a fleet and instance role
+ *   - A policy to create tags for spot fleet requests
+ *
+ * Resources Deployed
+ * ------------------------
+ * - An AWS Lambda that is used to connect to the render queue, and save Spot Event Plugin configurations.
+ * - A CloudFormation Custom Resource that triggers execution of the Lambda on stack deployment, update, and deletion.
+ * - An Amazon CloudWatch log group that records history of the AWS Lambda's execution.
+ * - An IAM Policy attached to Render Queue's Role.
+ *
+ * Security Considerations
+ * ------------------------
+ * - The AWS Lambda that is deployed through this construct will be created from a deployment package
+ *   that is uploaded to your CDK bootstrap bucket during deployment. You must limit write access to
+ *   your CDK bootstrap bucket to prevent an attacker from modifying the actions performed by this Lambda.
+ *   We strongly recommend that you either enable Amazon S3 server access logging on your CDK bootstrap bucket,
+ *   or enable AWS CloudTrail on your account to assist in post-incident analysis of compromised production
+ *   environments.
+ * - The AWS Lambda function that is created by this resource has access to both the certificates used to connect to the render queue,
+ *   and the render queue port. An attacker that can find a way to modify and execute this lambda could use it to
+ *   execute any requets against the render queue. You should not grant any additional actors/principals the ability to modify
+ *   or execute this Lambda.
+ */
+export class ConfigureSpotEventPlugin extends Construct {
+
+  /**
+   * Only one Spot Event Plugin Configuration is allowed per render queue / repository.
+   */
+  private static uniqueRenderQueues: Set<IRenderQueue> = new Set<IRenderQueue>();
+
+  constructor(scope: Construct, id: string, props: ConfigureSpotEventPluginProps) {
+    super(scope, id);
+
+    if (ConfigureSpotEventPlugin.uniqueRenderQueues.has(props.renderQueue)) {
+      throw new Error('Only one ConfigureSpotEventPlugin construct is allowed per render queue.');
+    }
+    else {
+      ConfigureSpotEventPlugin.uniqueRenderQueues.add(props.renderQueue);
+    }
+
+    if (props.renderQueue instanceof RenderQueue) {
+      // We do not check the patch version, so it's set to 0.
+      const minimumVersion: Version = new Version([10, 1, 12, 0]);
+
+      if (props.renderQueue.version.isLessThan(minimumVersion)) {
+        throw new Error(`Minimum supported Deadline version for ${this.constructor.name} is ` +
+        `${minimumVersion.versionString}. ` +
+        `Received: ${props.version.versionString}.`);
+      }
+
+      if (props.spotFleets && props.spotFleets.length !== 0) {
+        props.renderQueue.addSEPPolicies(props.configuration?.enableResourceTracker ?? true);
```

Comment:
If `enableResourceTracker` is set to `false`, then we won't add `AWSThinkboxDeadlineResourceTrackerAdminPolicy` to the RQ. But then customers can enable the resource tracker later, let's say in Monitor, and Resource Tracker will fail because of missing IAM permissions.
Do you think we should always add these policies, or if customers use RFDK for initial configuration, we can expect them to use RFDK for updats as well?

Reply:
Principal of least privilege dictates not adding unnecessary permissions. Usability suggests that always adding the resource tracker permissions would be a good idea.

Looking at the permissions in the managed policy... it looks like it's very tightly constrained. I don't think there's any harm in biasing towards usability here.